memo_app.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'sinatra/reloader'
+require 'json'
+require 'securerandom'
+
+set :show_exceptions, :after_handler
+
+# メモのインスタンスを作成し保存
+class BaseMemosData
+  attr_accessor :params
+
+  def initialize(params)
+    params['id'] = SecureRandom.uuid
+
+    created_at = Time.new
+    params['created_at'] = created_at.strftime('%Y年%m月%d日%H時%M分')
+    @params = params
+  end
+
+  def self.register(file_name, memo)
+    base_memos = JSON.parse(File.read(file_name))
+    base_memos << memo
+
+    BaseMemosData.update(file_name, base_memos)
+  end
+
+  def self.update(file_name, memo)
+    File.open(file_name, 'w') do |file|
+      JSON.dump(memo, file)
+    end
+  end
+end
+
+before '/memos/:id*' do
+  @memos = get_memos('memos.json')
+  @memos.each do |hash|
+    @memo_data = hash if hash['id'] == params['id']
+  end
+end
+
+helpers do
+  def request_accept?(url)
+    raise StandardError unless request.accept?(url)
+  end
+end
+
+helpers do
+  def get_memos(file_name)
+    @memos = JSON.parse(File.read(file_name))
+  end
+end
+
+helpers do
+  def h(text)
+    Rack::Utils.escape_html(text)
+  end
+end
+
+error StandardError do
+  puts "エラーが発生しました。 - #{env['sinatra.error'].message}"
+  erb :not_found
+end
+
+get '/memos' do
+  all_memos = get_memos('memos.json')
+  @all_memos = all_memos
```

Comment:
ここを２行に分ける必要はないので

```suggestion
  @all_memos = get_memos('memos.json')
```

これで良さそうです。

Reply:
こちら修正いたしました🙏